---
title: 'test microsat table'
author: "Maxime Dahirel"
date: "26/06/2020"
output: 
  html_document:
    theme: yeti
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,message=FALSE, warning=FALSE)
```
# S4 - 

## S4.1 - Design of the microsatellite markers and multiplex PCR assay

Microsatellite DNA libraries were produced following the method described in <!--REF TO ADD Malausa et al. (2011)-->. This method consists of (i) the isolation of DNA fragments containing microsatellite motifs by multiplex microsatellite enrichment (obtained via hybridisation to biotin-labelled oligonucleotides with the motifs (AG)~10~, (AC)~10~, (AAC)~8~, (AGG)~8~, (ACG)~8~, (AAG)~8~, (ACAT)~6~ and (ATCT)~6~) and (ii) the pyrosequencing of the fragments (454 GS-FLX Titanium). This protocol was applied to a sample containing DNA extracted from about 100 *Trichogramma brassicae* individuals.

The obtained DNA dataset was analyzed and primers for microsatellite amplification were designed using the QDD program <!--(Meglécz et al. 2010)-->, version 2.1. The default QDD parameters were used, except for: (i) the minimum percentage similarity of sequences used for the construction of consensus sequences was 90%, (ii) the proportion of sequences that must have the same base on the aligned site to accept it as a consensus was 0.66, (iii) the minimum length of PCR product for primer design was 80bp, (iv) the maximum length of PCR product for primer design was 500, (v) the maximum length of a primer was 32, (vi) the maximum acceptable difference between the melting temperatures of primers was 4 °C.

Nineteen microsatellite markers were developed by testing and multiplexing part of the PCR primers selected after the QDD analysis. The main criteria for microsatellites sequences selection and primers design used were (i) the motif of the microsatellite they target, to obtain a balanced frequency of motifs, (ii) microsatellites with at least eight repeats, (iii) the size of the expected PCR products (we selected primers amplifying markers homogenously distributed in terms of size within the range 80 bp - 500 bp).

Allelic richness of the markers and whether or not their allele frequencies followed the Hardy-Weinberg equilibrium was investigated by genotyping the initial pool of individuals <!-- to precise and change depending on idividuals in table-->  used in the experiment presented in this manuscript <!--(REF TO ADD Table SXX)-->.  

## S4.2 - Genotyping and microsatellites description  

All *T. brassicae* wasps were stored in 70% ethanol before DNA extraction. DNA was extracted from each individual wasp, either with the prepGem Insect DNA Extraction kit from Zygem (extraction volume of 15µL per individual, using 0.375µL of prepGem enzyme; samples were placed in a thermocycler for 2 hours at 75°C and 5 minutes at 95°C), or with the Quick Extract DNA Extraction Solution from Lucigen (extraction volume of 15µL per individual; samples were placed in a thermocycler for 10 minutes at 65°C and 2 minutes at 98°C). All DNA extracts were stored at -20°C.  

For microsatellite amplification, 2µl of genomic DNA were used within a 10µL final PCR reaction consisting of 5µL of the QIAGEN PCR multiplex solution, 1µL of the primer mix of the 19 primers reverse and forward (1µM each), and water. PCR conditions were as followed: initial denaturation at 95°C for 15 min; 30 cycles of denaturation (94°C, 30s), annealing (60°C, 90s) and elongation (72°C, 60s); final extension at 60°C for 30 min.
For each individual, 1.5µL of the PCR product was mixed with 9.8µL of a mix obtained from 1mL HiDi formamide and 20µL of 500 LIZ Size Standard (Applied Biosystems). Products were then electrophoresed using an ABI PRISM 3130XL sequencer (Applied Biosystems). Finally, genotypes were scored using Gene Marker version 1.75 (SoftGenetics).

**Supplementary table S4.1**: Primer pairs used for the PCR set of the multiplex panel. Population genetics statistics presented here were computed with <!--CHANGE IF DONE WITH R, the software GenePop version 3.3 REF TO ADD (Raymond & Rousset 1995, Heredity 86, 248-249)--> from the<!-- ADAPT SET full set of individuals of genetic mix 1  at the start of the experiment-->. For each of the 19 loci, the table includes: the repeat motifs in the sequence used to design primers, sequence and fluorescent dye label used for each primer, <!--number of individuals successfully genotyped, observed number of alleles, allelic size range, observed heterozygosity (HO), expected heterozygosity (HE, Nei 1978, Genetics 89, 589-590) and P-value for the exact test of Hardy–Weinberg equilibrium (HWE). Note that all loci were found to be polymorphic when analysing the whole set of wasps used in the experiment (i.e. genetic mixes 1, 2 and 3)-->. 

<!--add supplementary table-->

```{r}
library(tidyverse)
library(kableExtra)
library(here)

library(adegenet)

raw_genetics <- read_csv(here("data/Trichogramma_genetics.csv"), 
                         col_types = cols(.default=col_character())
)

data_genetics<-raw_genetics %>% 
  mutate( ### info on variables is contained in each sample ID
    Generation=str_extract(ID,"G0|G4|G8|G12"),
    Mix=str_extract(ID,"_1P|_2P|_3P")
  ) %>% 
  mutate(
    Generation=as.numeric(str_extract(Generation,"0|4|8|12")),
    Mix=as.numeric(str_extract(Mix,"1|2|3"))) #%>% 
  #filter(Generation==0)

genmat<- df2genind(select(data_genetics,starts_with("TB")),####all allelic columns start with a "P"
            ncode=3,
            NA.char = "000", ## missing alleles
            ind.names=data_genetics$ID,pop=data_genetics$Mix) #ncode=3 bc 3 characters for one allele

#pegas::hw.test(seppop(genmat)$`1`) careful, rows with df = 0 = rows with 0 diversity, convert to NA/--
# nAll(genmat) do allele counts and seq range on total data (some locus are monomorphic on G0 because sampling)
# summary(genmat)$Hexp
# summary(genmat)$Hobs

test=enframe(genmat@all.names) %>% 
  unnest(cols=value) %>% 
  mutate(value=as.numeric(value)) %>% 
  filter(value>0) %>% ##remove missing alleles (scored as 0s)
  group_by(name) %>% 
  summarise(min=min(value),max=max(value)) %>% 
  mutate(range=paste(min,"-",max)) %>% 
  left_join(tibble(nAll=nAll(genmat),name=names(nAll(genmat)))) %>% 
  select(Locus = name,`number of alleles`=nAll,`Size range (bp)`=range)

read_csv(here("data","Trichogramma_microsatellite_description.csv")) %>%
  rename(Locus="locus",`Repeat motif`="motif",Forward="forward_primer",Reverse="reverse_primer",`Fluorescent dye`="dye") %>% 
  left_join(test) %>% 
  knitr::kable(format="html") %>%
  kable_styling() %>% 
  add_header_above(c(" " = 2, "Primers sequence (5'-3')" = 2, " "=3)) %>% 
  collapse_rows(5,valign="top") ##
```


